Update iOS Llama demo app readme docs

Riandy opened the pull request:
Summary:
- Revamp and standardize the readme docs structure for better clarity.
- Added different delegate instructions for ios.
- Added some screenshots to improve readme clarity.

Differential Revision: D62660106

jeffxtang commented on examples/demo-apps/apple_ios/LLaMA/docs/delegates/xnnpack_README.md:
````diff
@@ -0,0 +1,107 @@
+Export the model
+```
+python -m examples.models.llama2.export_llama --checkpoint <consolidated.00.pth> -p <params.json> -kv --use_sdpa_with_kv_cache -X -qmode 8da4w  --group_size 128 -d fp32 --metadata '{"get_bos_id":128000, "get_eos_ids":[128009, 128001]}' --embedding-quantize 4,32 --output_name="llama3_kv_sdpa_xnn_qe_4_32.pte"
+```
````
Do we have instructions on how to export Llava?

Riandy committed:
Update iOS Llama demo app readme docs (#5359)

Summary:
Pull Request resolved: https://github.com/pytorch/executorch/pull/5359

- Revamp and standardize the readme docs structure for better clarity.
- Added different delegate instructions for ios.
- Added some screenshots to improve readme clarity.

Reviewed By: larryliu0820

Differential Revision: D62660106
